libpimeval/src/libpimeval.h

```diff
@@ -147,16 +147,17 @@ PimStatus pimMaxScalar(PimObjId src, PimObjId dest, uint64_t scalarValue);
 // multiply src1 with scalarValue and add the multiplication result with src2. Save the result to dest. 
 PimStatus pimScaledAdd(PimObjId src1, PimObjId src2, PimObjId dest, uint64_t scalarValue);
 PimStatus pimPopCount(PimObjId src, PimObjId dest);
-PimStatus pimRedSumInt(PimObjId src, int64_t* sum);
-PimStatus pimRedSumUInt(PimObjId src, uint64_t* sum);
-PimStatus pimRedSumFP(PimObjId src, float* sum);
+PimStatus pimRedSum(PimObjId src, void* min, uint64_t idxBegin = 0, uint64_t idxEnd = 0);
+// Min/Max Reduction APIs
+PimStatus pimRedMin(PimObjId src, void* min, uint64_t idxBegin = 0, uint64_t idxEnd = 0);
+PimStatus pimRedMax(PimObjId src, void* max, uint64_t idxBegin = 0, uint64_t idxEnd = 0);
+
 // Note: Reduction sum range is [idxBegin, idxEnd)
-PimStatus pimRedSumRangedInt(PimObjId src, uint64_t idxBegin, uint64_t idxEnd, int64_t* sum);
-PimStatus pimRedSumRangedUInt(PimObjId src, uint64_t idxBegin, uint64_t idxEnd, uint64_t* sum);
-PimStatus pimRedSumRangedFP(PimObjId src, uint64_t idxBegin, uint64_t idxEnd, float* sum);
+
+PimStatus pimRedSumRangedFP32(PimObjId src, uint64_t idxBegin, uint64_t idxEnd, float* sum);
```

Comment:
This can be removed

Reply:
removing this caused an error:
src/pimCore.cpp:35:17: error: ‘PIM_RREG_R2’ was not declared in this scope; did you mean ‘eIM_RREG_R2’?
   35 |   declareRowReg(PIM_RREG_R2);
      |                 ^~~~~~~~~~~
      |                 eIM_RREG_R2
src/pimCore.cpp: In constructor ‘pimCore::pimCore(unsigned int, unsigned int)’:
src/pimCore.cpp:35:17: error: ‘PIM_RREG_R2’ was not declared in this scope; did you mean ‘eIM_RREG_R2’?
   35 |   declareRowReg(PIM_RREG_R2);
      |                 ^~~~~~~~~~~
      |                 eIM_RREG_R2